clang/lib/Sema/SemaType.cpp

```diff
@@ -1500,11 +1500,14 @@ static QualType ConvertDeclSpecToType(TypeProcessingState &state) {
       Result = Context.DoubleTy;
     break;
   case DeclSpec::TST_float128:
-    if (!S.Context.getTargetInfo().hasFloat128Type() && 
-        !S.getLangOpts().SYCLIsDevice &&
-        !(S.getLangOpts().OpenMP && S.getLangOpts().OpenMPIsDevice))
+    if (S.getLangOpts().SYCLIsDevice)
```

Comment:
```suggestion
    if (!S.Context.getTargetInfo().hasFloat128Type() && S.getLangOpts().SYCLIsDevice)
```
This seems to be a more accurate check. There might be targets supporting `__float128` in AOT mode and using it should be okay in this case. Agree?

Reply:
Okay, agree

Reply:
Done.